app/user_api/User.py

```diff
@@ -29,29 +29,61 @@ def __init__(
         self.hasCompletedOnboarding = hasCompletedOnboarding
 
     def get(db_ref, id):
+        """
+        Fetches a specified user from a database.
+
+        Args:
+            db_ref: A `CollectionReference` to the users table from Firestore
+            id: A string of the user email to fetch
+
+        Returns:
+            User: specified by `id`
+
+        Raises:
+            UserNotFoundError: if the target user does not exist
+        """
         target_data = db_ref.document(id).get()
         if not target_data.exists:
             raise UserNotFoundError(id)
         return User.from_dict(db_ref, target_data.to_dict())
 
     def from_dict(db_ref, dict):
+        """
+        Creates a new User object from a Python Dictionary
+
+        Args:
+            db_ref: A `CollectionReference` to the users table from Firestore
+            dict: Dictionary of key-value pairs corresponding to user.
+
+        Returns:
+            User: from specified data
+
+        Raises:
+            BadDataError: If required data is missing from the dictionary
+        """
         try:
             return User(
                 db_ref,
                 dict["email"],
-                dict["referral_code"],
-                dict["reward_point_balance"],
-                dict["notification_setting"],
-                dict["color_theme"],
-                dict["time_in_line"],
-                dict["num_lines_participated"],
-                dict["poi_frequency"],
-                dict["hasCompletedOnboarding"],
+                dict.get("referral_code", ""),
+                dict.get("reward_point_balance", 0),
+                dict.get("notification_setting", False),
+                dict.get("color_theme", "SYSTEM"),
```

Comment:
This is stored natively on their phone, so no need for it in the database.

Reply:
Got it, I'll remove this